capstone-project/src/main/angular-webapp/src/app/authentication/authentication.component.ts

```diff
@@ -1,31 +1,37 @@
 import { Component, OnInit } from '@angular/core';
-import { SocialAuthService } from "angularx-social-login";
-import { GoogleLoginProvider } from "angularx-social-login";
-import { SocialUser } from "angularx-social-login";
+import { SocialAuthService, GoogleLoginProvider } from "angularx-social-login";
+import { UserService } from "../user.service";
 
 @Component({
   selector: 'app-authentication',
   templateUrl: './authentication.component.html',
   styleUrls: ['./authentication.component.css']
 })
 export class AuthenticationComponent implements OnInit {
-
-  user: SocialUser;
-
-  constructor(private authService: SocialAuthService) { }
+  
+  constructor(private authService: SocialAuthService, private userService: UserService) { }
 
   ngOnInit(): void {
+    // When user status changes update the user service
     this.authService.authState.subscribe((user) => {
-      this.user = user;
+      this.userService.currentUser = user;
     });
   }
 
+  // Sign in user
   signInWithGoogle(): void {
     this.authService.signIn(GoogleLoginProvider.PROVIDER_ID);
   }
 
+  // Sign out user and reload page
   signOut(): void {
     this.authService.signOut();
+    window.location.reload();
+  }
+
+  // Return the current user
+  getUser(){
```

Comment:
Try:

get user(): SocialUser {
  return this.userService.currentUser;
}

I think then you can access user in HTML as before (just with 'user' and not 'user()')

Reply:
Cool! Thanks